rust/ruby-prism/build.rs

```diff
@@ -304,8 +305,13 @@ fn write_node(file: &mut File, flags: &[Flags], node: &Node) -> Result<(), Box<d
                 writeln!(file, "    }}")?;
             },
             NodeFieldType::String => {
-                writeln!(file, "    pub const fn {}(&self) -> &str {{", field.name)?;
-                writeln!(file, "        \"\"")?;
+                writeln!(file, "    pub fn {}(&self) -> &str {{", field.name)?;
+                writeln!(file, "        unsafe {{")?;
+                writeln!(file, "            let source = (*self.pointer).{}.source;", field.name)?;
+                writeln!(file, "            let length = (*self.pointer).{}.length;", field.name)?;
+                writeln!(file, "            let slice = std::slice::from_raw_parts(source, length);")?;
+                writeln!(file, "            std::str::from_utf8_unchecked(slice)")?;
```

Comment:
Yeah we can't do this. Strings can be contents of regexps, xstrings, strings, and symbols, all of which can have any of the 90 source encodings that Ruby supports. Better to return a u8 slice and have the consumer handle it.